packages/yoastseo/src/languageProcessing/researches/findKeywordInFirstParagraph.js

```diff
@@ -42,12 +42,16 @@ export default function( paper, researcher ) {
 	const topicForms = researcher.getResearch( "morphology" );
 	const matchWordCustomHelper = researcher.getHelper( "matchWordCustomHelper" );
 	const locale = paper.getLocale();
+	const startOffset = firstParagraph && firstParagraph.sourceCodeLocation.startOffset;
 
+	const mappedBlocks = paper._attributes.wpBlocks;
+	const filteredIntroductionBlock = mappedBlocks && mappedBlocks.filter( block => inRange( startOffset, block.startOffset, block.endOffset ) )[ 0 ];
 	const result = {
 		foundInOneSentence: false,
 		foundInParagraph: false,
 		keyphraseOrSynonym: "",
 		introduction: firstParagraph,
+		parentBlock: filteredIntroductionBlock || null,
```

Comment:
Here I also return the parent block of the introduction node. This information will be used for sending the correct introduction text for the Yoast AI Optimize, when the introduction is part of a classic block.